### PR TITLE
Reboot RS5 after test run if there is less than 15 GB free disk space

### PR DESCRIPTION
**Pull request by olljanat**
Added reboot for Windows RS5 nodes which happens 15 seconds after CI build is finalized if there is less than 15 GB space left on D: -drive.

One more workaround to #38114

### CODE REVIEW

**Comment by olljanat** [comment]
@thaJeztah @kolyshkin @johnstep it looks that RS5 builds have been now working after I managed to reboot them ( details are on https://github.com/moby/moby/issues/38114#issuecomment-448370336 ) so it would be nice to keep them on working (earlier they was all time failing to error like on this one https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS5-Process/641/console )

@jhowardmsft looks to be away so maybe you need to so if we can add this kind of temporary solution here?

**Comment by olljanat** [comment]
Derek add label: rebuild/powerpc
Derek add label: rebuild/experimental

**Comment by olljanat** [comment]
Derek add label: rebuild/experimental

**Comment by olljanat** [comment]
Derek add label: rebuild/powerpc

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Comment by olljanat** [comment]
Derek add label: rebuild/*

**Comment by olljanat** [comment]
Derek add label: rebuild/z

File: hack/ci/windows.ps1 [hunk 1]
```diff
@@ -994,4 +994,14 @@ Finally {
     Nuke-Everything
     $Dur=New-TimeSpan -Start $StartTime -End $(Get-Date)
     Write-Host -ForegroundColor $FinallyColour "`nINFO: executeCI.ps1 exiting at $(date). Duration $dur`n"
+
+    # Temporary solution to avoid disk to fill on RS5
+    if ($env:COMPUTERNAME -match "jenkins-rs5-") {
+		if ((Get-Volume -DriveLetter D).SizeRemaining -lt 15GB) {
+			Write-Host -ForegroundColor Green "INFO: D -drive is almost full, rebooting..."
+			shutdown -r -t 15
```
Comment (johnstep) [inline hack/ci/windows.ps1]: Nit: This is fine but could use the `Restart-Computer` PowerShell command, though the delay would be done separately, e.g. `Start-Sleep 15; Restart-Computer`, if waiting is important.

File: hack/ci/windows.ps1 [hunk 2]
```diff
@@ -994,4 +994,14 @@ Finally {
     Nuke-Everything
     $Dur=New-TimeSpan -Start $StartTime -End $(Get-Date)
     Write-Host -ForegroundColor $FinallyColour "`nINFO: executeCI.ps1 exiting at $(date). Duration $dur`n"
+
+    # Temporary solution to avoid disk to fill on RS5
+    if ($env:COMPUTERNAME -match "jenkins-rs5-") {
+		if ((Get-Volume -DriveLetter D).SizeRemaining -lt 15GB) {
+			Write-Host -ForegroundColor Green "INFO: D -drive is almost full, rebooting..."
```
Comment (johnstep) [inline hack/ci/windows.ps1]: Nit: Remove the `-`, just `D drive` (or `$env:TESTRUN_DRIVE drive`).

**Review by johnstep — APPROVED** [review]
It is okay as is, but see the suggestions.

**Comment by olljanat** [comment]
@johnstep update based on review. Except changing that shutdown command as it is there on purpose to make sure that build can pass before server reboots

**Comment by olljanat** [comment]
Derek add label: rebuild/z

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!!

**Comment by jhowardmsft** [comment]
No, please do not do this. It has other implications which go back to TP4 days. Jenkins often schedules another job immediately on a node, and rebooting it will cause that subsequent job to fail when the SSH connection drops. We are aware of what the issue is (it's a kernel issue), and this only masks it, not fixes it. Let's get the right fix in instead in a subsequent Windows update. I have been on vacation since 9th Dec, and won't be back until the new year, but will try to get the servers back to working again this week which should suffice as an interim measure. Hopefully tomorrow.

**Comment by olljanat** [comment]
@jhowardmsft How about scheduled script to those servers which runs once of day, check that there is no jobs running and reboot server?

Anyone who have access those servers can create that easily. 

It does not hide issue from logs so we can see when windows update fixes the issue and error disappears

Alternatively if you can provide access to those servers for someone else too who can reboot them when they are out of disk then we can live with situation.

PS. I will close this PR now but we can continue discussion here (or Slack)

**Comment by olljanat** [comment]
@jhowardmsft FYI, at least jenkins-rs5-3 looks to be now out of disk so it would be nice to get some temporary solution in-place.

**Comment by olljanat** [comment]
and now disk was full on all three RS5 servers so I used this PR (once again) to reboot them.

**Comment by jhowardmsft** [comment]
@olljanat @thaJeztah @johnstep We think we might have a fix. I'm going to re-deploy these machines in the next 24 hrs and hopefully all will be good again.

**Pull request closed without merging**